lib/py/test/test_sslsocket.py

```diff
@@ -75,185 +104,176 @@ def __exit__(self, exc_type, exc_value, traceback):
 
 
 class TSSLSocketTest(unittest.TestCase):
-    def _assert_connection_failure(self, server, client):
+    def _server_socket(self, **kwargs):
+        return TSSLServerSocket(port=0, **kwargs)
+
+    @contextmanager
+    def _connectable_client(self, server, path=None, **client_kwargs):
         acc = ServerAcceptor(server)
         try:
             acc.start()
-            time.sleep(CONNECT_DELAY / 2)
-            client.setTimeout(CONNECT_TIMEOUT / 2)
-            with self._assert_raises(Exception):
-                logging.disable(logging.CRITICAL)
-                client.open()
-                select.select([], [client.handle], [], CONNECT_TIMEOUT / 2)
-            # self.assertIsNone(acc.client)
-            self.assertTrue(acc.client is None)
+            acc.await_listening()
```

Comment:
Ah, I think I finally got it.
Maybe you mean `self._listening.wait(timeout=timeout)` rather than `_port_bound` here ?
I missed your intention of waiting for `listen()` rather than `getsockname()`
Could you change it to _listening or remove the _listening that is unused ?